src/main/kotlin/org/gradle/script/lang/kotlin/provider/CachingKotlinCompiler.kt

```diff
@@ -94,16 +94,17 @@ class CachingKotlinCompiler(
 
     fun compileBuildScript(
         scriptFile: File,
+        script: String,
         additionalSourceFiles: List<File>,
         classPath: ClassPath,
         parentClassLoader: ClassLoader): CompiledScript {
 
         val scriptFileName = scriptFile.name
-        return compileScript(cacheKeyPrefix + scriptFileName + scriptFile, classPath, parentClassLoader) {
+        return compileScript(cacheKeyPrefix + scriptFileName + script, classPath, parentClassLoader) { cacheDir ->
```

Comment:
Yep, performance wise the difference is that before we hashed the file from disk and now we hash its content, that was already loaded in any case, from memory.